### PR TITLE
Interchangeable Spell Script

### PR DESCRIPTION
**Pull request by MatthewFrench**
Modified spell scripts to be more generic and interchangeable between passive, spells and buffs(and eventually gameobject AI). Blanked out all empty spells.

### CODE REVIEW

File: Buffs/LuluR/LuluR.cs [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
         float healthBefore;
         float meantimeDamage;
         float healthNow;
         float healthBonus;
-
-        public void OnActivate(Unit unit, Spell ownerSpell)
+        public void OnActivate(GameScriptInformation scriptInfo)
         {
+            info = scriptInfo;
+            spell = info.OwnerSpell;
```
Comment (MythicManiac) [inline Buffs/LuluR/LuluR.cs:21]: These should be implemented in a way that we don't need to re-declare them in every spell. Maybe add a property to a base class

**Review by MythicManiac — CHANGES_REQUESTED** [review]
It feels silly to re-declare the same variables in every file, a way to inherit those properties should be looked into.

File: Buffs/LuluR/LuluR.cs (hunk 1 above)
Comment (MythicManiac) [reply Buffs/LuluR/LuluR.cs:21]: Adding to this: it cannot be done with an interface, but if an interface isn't necessary we might want to just roll with abstract. Optionally we could add an abstract class that extends from the interface if a scenario like that would make more sense.

For example:
`IGameScript`
`AbstractGameScript : IGameScript`
`LuluR : AbstractGameScript`

Whatever is the most simple and flexible solution should be used, explore some options :)

**Comment by moonshadow565** [comment]
public string Foo { get; set; }
stuff like this can be used in interfaces

**Comment by MythicManiac** [comment]
Using properties in interfaces won't implement them, it will just make it required for the subclasses to implement them AFAIK

**Comment by MythicManiac** [comment]
Closing this as stale (almost a year old and needs huge rebase)

**Pull request closed without merging**